File: src/TrafficLight.cpp
#include "TrafficLight.h"

TrafficLight::TrafficLight(std::string approachId) : approachId(std::move(approachId)) {}

LightState TrafficLight::getState() const { return state; }

void TrafficLight::setState(LightState newState) { state = newState; }

const std::string &TrafficLight::getApproachId() const { return approachId; }

File: python/cross_roads_ai/traffic_light.py
from .light_state import LightState

class TrafficLight:
    def __init__(self, approachId: str):
        self._approachId = approachId
        self._state = LightState.Red

    def getState(self) -> LightState:
        return self._state

    def setState(self, state: LightState) -> None:
        self._state = state

    def getApproachId(self) -> str:
        return self._approachId
